flaps/flaps.go

```diff
@@ -204,6 +214,38 @@ func (f *Client) _sendRequest(ctx context.Context, method, endpoint string, in,
 	return nil
 }
 
+type flapsCall struct {
+	Call       string  `json:"c"`
+	Command    string  `json:"co"`
+	StatusCode int     `json:"s"`
+	Duration   float64 `json:"d"`
+}
+
+func sendFlapsCallMetric(ctx context.Context, endpoint flapsAction, statusCode int, duration time.Duration) {
+	cmdCtx := command_context.FromContext(ctx)
+
+	// Iterate backwards through the command name to figure out the command being run.
+	// For example, `fly m run` becomes `machine-run`, `deploy --flags` becomes `deploy`
+	// TODO(billy): Unit test this
+	var nameParts []string
+	for cmdCtx != nil {
+		// Don't include the binary name in the nameParts
+		if cmdCtx.Name() == "flyctl" {
```

Comment:
haven't tested it, does it work when the user calls flyctl through the `fly` alias?

Reply:
This is a good point, I didn't consider other binary names.